### DISCUSSION TITLE
Onboarding

### DISCUSSION THREAD

**Discussion by henriqueweege** [AvaloniaUI/Avalonia#12746] [General]
Hi guys! Hope everyone is doing great!
As I stated in the telegram group some days ago, I would really love to contribute to this project through coding, but until now, I am feeling quite lost with everything, and although I did look into the documentation, I was unable to find something that explained about the structure of the project or how it works in depth. I would like to know: there is something like that? I really think that this would be of great help to make the process of knowing the platform easier.
Please, if you have any suggestions on how I can familiarize with it, let me know!

Cheers,
Henrique

**Comment by timunie** [comment]
check out src/samples ^^ Other than that, nothing I'm aware of.

**Reply by henriqueweege** [reply]
@timunie  thanks!!

**Comment by stevemonaco** [comment]
~~There's no enforced pattern as project and folder structure is an opinion. Projects with different architectures and different supported platforms will necessarily organize themselves a bit differently. Same if you have multiple clients (eg. Avalonia and a standalone console app), use a pattern other than MVVM, use feature folders instead of separate folders for Views and ViewModels, etc.~~

~~You might check out some of these projects for comparison: [ChatGPT Client](https://github.com/wieslawsoltes/ChatGPT), [WalletWasabi](https://github.com/zkSNACKs/WalletWasabi), [TileShop](https://github.com/stevemonaco/ImageMagitek) for ideas that might fit your project.~~

**Reply by timunie** [reply]
@stevemonaco the question was about how to contribute to avalonia unless I have overseen something

**Reply by stevemonaco** [reply]
I may have misunderstood the aim of the question as being an intermediate step of how to use Avalonia (well) in applications in preparation of eventually contributing to Avalonia itself.

**Reply by henriqueweege** [reply]
@timunie and @stevemonaco  thanks for your replies.
Maybe I wasn't clear, and I am sorry for that, let me try to explain myself. 
The problem I am facing is that, although I did clone the project, looked into it, and ran it as the docs show, I am unable to grasp how the parts interact with each other. So, for example, the project "_build", the name indicate that this project have something to do with the build process, but following the docs I needed to build another project to run Avalonia. This lack of understanding of how the parts work together to make the whole, is something that I need to overpass to be able to successfully contribute to the project.
My question is, besides looking into src/samples, as @timunie spotted, there is another ways to fulfill this gaps of knowledge?

**Reply by stevemonaco** [reply]
There's an in-progress [contributing PR](https://github.com/AvaloniaUI/Avalonia/pull/12714) from a few days ago. You might check that out. For building, there's an [article](https://github.com/AvaloniaUI/Avalonia/blob/master/Documentation/build.md) on the wiki. It's pretty disjointed, so if a team member doesn't correct me, you might want to create an issue about consolidating these resources and to document the major project components for new contributors.

**Reply by henriqueweege** [reply]
I really think that a more structured onboarding process would be of great help for newbies as myself, so I will follow your advice. I will just wait a little bit to give anyone the possibility of reply to this thread before I open an issue.
Thanks a lot!!

**Reply by timunie** [reply]
Talked to the team just now. We have:

new: https://avaloniaui.net/Blog/avalonia-platform-support-why-it-s-simple,ca378b03-fb47-4c34-a347-182acaa43d05

older: https://github.com/AvaloniaUI/Avalonia/wiki/High-level-architecture-overview

But probably it's not 100% well enough for beginners. So probably you can note down what was missing for you while trying to understand the code base and help us to make onboarding better?

**Reply by henriqueweege** [reply]
@timunie thank you very much!!
For sure I will do that. Will be nice to be able to contribute in some way!
Thanks again!!
